### DISCUSSION TITLE
Use the latest code version with an existing db?

### DISCUSSION THREAD

**Discussion by winzkigermany** [inventree/InvenTree#4607] [Q&A] [answered]
I'd like to try some of the latest features from the master branch, for example by using the docker development installation according to
https://docs.inventree.org/en/latest/start/docker_dev/
What's the best way to use my existing database from a production installation with it, so I can "git pull", "docker compose" from time to time, and keep the existing database?

**Comment by matmair** [comment]
You can work with the latest tag; We can not and do not guarantee that that won't brick your installation. So probably do not use that for production if you do not have reliable backups and set up a separate database for testing.

**Comment by SchrodingersGat** [comment]
> What's the best way to use my existing database from a production installation with it, so I can "git pull", "docker compose" from time to time, and keep the existing database?

If you are running the development docker image, it is very straight forward to keep updated:


https://docs.inventree.org/en/latest/start/docker_dev/#database-updates

**Reply by winzkigermany** [reply, marked as answer]
Thank you.

I've chosen a half- and half approach:
I'm using a local git clone of the repo, and I modified docker-compose.yml to build from there.
So I can work with a stable tag, and still try out some code from time to time.

Not full-dev and not full production. Works for me (tm)
As a fallback, all of it runs on a vmware guest, so I can roll back to a known-good snapshot, just in case.
